compliance_checker/cf/cf_1_7.py

```diff
@@ -749,33 +841,29 @@ def check_grid_mapping(self, ds):
                                 str(crs_error)
                             )
                         )
-                    else:
-                        test_ctx.score += 1
-                    test_ctx.out_of += 1
-
+                        test_ctx.out_of += 1                                   
+            test_ctx.score += 1
+ 
             # existence_conditions
+            # Both geoid_name and geopotential_datum_name cannot exist
             exist_cond_1 = (
                 self._check_gmattr_existence_condition_geoid_name_geoptl_datum_name(var)
             )
-            test_ctx.assert_true(exist_cond_1[0], exist_cond_1[1])
+            test_ctx.messages.append(exist_cond_1)
+            test_ctx.score += 1
+
+            # [8/9] reference_ellipsoid_name, prime_meridian_name, horizontal_datum_name and 
+            # geographic_crs_name must be all defined if any one is defined.
             exist_cond_2 = self._check_gmattr_existence_condition_ell_pmerid_hdatum(var)
-            test_ctx.assert_true(exist_cond_2[0], exist_cond_2[1])
+            test_ctx.messages.append(exist_cond_2)
+            test_ctx.score += 1               
 
             # handle vertical datum related grid_mapping attributes
             vert_datum_attrs = {}
             possible_vert_datum_attrs = {"geoid_name", "geopotential_datum_name"}
-            vert_datum_attrs = possible_vert_datum_attrs.intersection(var.ncattrs())
-            len_vdatum_name_attrs = len(vert_datum_attrs)
-            # check that geoid_name and geopotential_datum_name are not both
-            # present in the grid_mapping variable
-            if len_vdatum_name_attrs == 2:
-                test_ctx.out_of += 1
-                test_ctx.messages.append(
-                    "Cannot have both 'geoid_name' and "
-                    "'geopotential_datum_name' attributes in "
-                    "grid mapping variable '{}'".format(var.name)
-                )
-            elif len_vdatum_name_attrs == 1:
+            vert_datum_attrs = possible_vert_datum_attrs.intersection(var.ncattrs())                             
+            
+            if  exist_cond_1[0] == True and vert_datum_attrs:              
```

Comment:
Just use `if exist_cond_1[0] and vert_datum_attrs` if `exist_cond[0]` is expected to be boolean.